rules/onchain_budget_test.go

```diff
@@ -0,0 +1,485 @@
+
+	values := &OnChainBudget{
+		AbsoluteAmtSats: 100,
+		MaxSatPerVByte:  40,
+	}
+
+	tx := newMockKVStoresTx()
+	cfg := &ConfigImpl{
+		Stores:    &mockKVStores{tx: tx},
+		ReqID:     0,
+		LndConnID: "test",
+	}
+
+	mgr := &OnChainBudgetMgr{}
+
+	enf := &OnChainBudgetEnforcer{
+		onChainBudgetConfig: cfg,
+		OnChainBudget:       values,
+		OnChainBudgetMgr:    mgr,
+	}
+
+	assertMemoInvariant := func(memo string) {
+		t.Helper()
+		require.Equal(t, memo, removeReqId(memo))
+	}
+
+	testMemo1 := "onBudget-abcdefghijklmnop-0: some"
+	clearMemo1 := removeReqId(testMemo1)
+	require.Equal(t, clearMemo1, " some")
+
+	testMemo2 := "some"
+	clearMemo2 := removeReqId(testMemo2)
+	require.Equal(t, clearMemo2, "some")
+
+	// We check that the memo field is correctly removed, such that a second
+	// removal leads to the same string. The detailed mechanics of
+	// removeReqId is tested in a separate test.
+	response, err := enf.HandleResponse(
+		ctx, "/lnrpc.Lightning/ListChannels",
+		&lnrpc.ListChannelsResponse{
+			Channels: []*lnrpc.Channel{
+				{Memo: testMemo1},
+				{Memo: testMemo2},
+			},
+		},
+	)
+	require.NoError(t, err)
+	for _, channel := range response.(*lnrpc.ListChannelsResponse).Channels {
+		assertMemoInvariant(channel.Memo)
+	}
+
+	response, err = enf.HandleResponse(
+		ctx, "/lnrpc.Lightning/PendingChannels",
+		&lnrpc.PendingChannelsResponse{
+			TotalLimboBalance: 0,
+			PendingOpenChannels: []*lnrpc.PendingChannelsResponse_PendingOpenChannel{
+				{
+					Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+						Memo: testMemo1,
+					},
+				},
+				{
+					Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+						Memo: testMemo2,
+					},
+				},
+			},
+			PendingClosingChannels: []*lnrpc.PendingChannelsResponse_ClosedChannel{
+				{
+					Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+						Memo: testMemo1,
+					},
+				},
+			},
+			PendingForceClosingChannels: []*lnrpc.PendingChannelsResponse_ForceClosedChannel{
+				{
+					Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+						Memo: testMemo1,
+					},
+				},
+			},
+			WaitingCloseChannels: []*lnrpc.PendingChannelsResponse_WaitingCloseChannel{
+				{
+					Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+						Memo: testMemo1,
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+	pending := response.(*lnrpc.PendingChannelsResponse)
+	for _, channel := range pending.PendingOpenChannels {
+		assertMemoInvariant(channel.Channel.Memo)
+	}
+
+	for _, channel := range pending.PendingClosingChannels {
+		assertMemoInvariant(channel.Channel.Memo)
+	}
+
+	for _, channel := range pending.PendingForceClosingChannels {
+		assertMemoInvariant(channel.Channel.Memo)
+	}
+
+	for _, channel := range pending.WaitingCloseChannels {
+		assertMemoInvariant(channel.Channel.Memo)
+	}
+}
+
+// TestRemoveMemo tests that request identifiers are correcly removed from the
+// memo string.
+func TestRemoveMemo(t *testing.T) {
+	tests := []struct {
```

Comment:
If you implement my suggestion above, I'd add another test that shows that you can have another part before the `onBudget-connID-reqID` part.
If you decide not to do it, I'd probably add another test to show that it's intended behaviour that we can't have another part before it currently.